### PR TITLE
Include Cython-AFF3CT example documentation.

### PR DESCRIPTION
**Pull request by SubtleMuffin**
This PR introduces the accompanying documentation of `Cython-AFF3CT` [example](https://github.com/aff3ct/my_project_with_aff3ct/pull/7).

### CODE REVIEW

**Comment by kouchy** [comment]
Thank you for this complete PR. I have to take the time to check the code. I will do that in the next days.

File: doc/source/user/library/examples.rst [hunk 1]
```diff
@@ -966,3 +966,318 @@ modules while in ``u.modules_stats`` the two dimension are switched.
...
+	    library_dirs=["../../lib/aff3ct/build/lib"],
+	    include_dirs=["../../lib/aff3ct/include", "../../lib/aff3ct/lib/cli/src", "../../lib/aff3ct/lib/MIPP/src", "../../lib/aff3ct/lib/MIPP/src", "../../lib/aff3ct/lib/rang/include"],
+	    language="c++",
+	    extra_compile_args=["-std=c++14"],
+	    extra_link_args=["-std=c++14"]
+	)
+	setup(
+	    name="codec",
+	    ext_modules=cythonize([codec])
+	)
```
Comment (kouchy) [inline doc/source/user/library/examples.rst:1274]: I think that we can remove this code, this is a configuration code.

File: doc/source/user/library/examples.rst [hunk 2]
```diff
@@ -966,3 +966,318 @@ modules while in ``u.modules_stats`` the two dimension are switched.
...
+
+	  // encode
+	  module::Encoder_polar<int> polar_encoder(k, n, frozen_bits);
+	  for (auto frame : info_bits) {
+	    std::vector<int> encoded(n);
+	    polar_encoder.encode(frame, encoded);
+	    encoded_bits.push_back(encoded);
+	  }
+	  return encoded_bits;
+	}
```
Comment (kouchy) [inline doc/source/user/library/examples.rst:1199]: I think we can remove this code.

File: doc/source/user/library/examples.rst [hunk 3]
```diff
@@ -966,3 +966,318 @@ modules while in ``u.modules_stats`` the two dimension are switched.
...
+
+	  // encode
+	  module::Decoder_polar_SC_naive<int> polar_decoder(k, n, frozen_bits);
+	  for (auto frame : received) {
+	    std::vector<int> decoded(k);
+	    polar_decoder.decode_siho(frame, decoded);
+	    decoded_bits.push_back(decoded);
+	  }
+	  return decoded_bits;
+	}
```
Comment (kouchy) [inline doc/source/user/library/examples.rst:1247]: I think we can remove this code too.

File: doc/source/user/library/examples.rst [hunk 4]
```diff
@@ -966,3 +966,318 @@ modules while in ``u.modules_stats`` the two dimension are switched.
 
 .. note:: The full source code is available here:
   https://github.com/aff3ct/my_project_with_aff3ct/blob/master/examples/openmp/src/main.cpp.
+
+
+Cython-AFF3CT
```
[the remark this answers is not in the document]
Comment (SubtleMuffin) [reply doc/source/user/library/examples.rst:971]: Done.

File: doc/source/user/library/examples.rst [hunk 5]
```diff
@@ -966,3 +966,318 @@ modules while in ``u.modules_stats`` the two dimension are switched.
...
+=============
+
+.. _Cython: https://cython.org/
+
+This example serves as a template framework on how to use AFF3CT in Python. 
+This example is based on the `Cython`_, and steps of compiling this example
+is included in the `README` file.
+
```
[the remark this answers is not in the document]
Comment (SubtleMuffin) [reply doc/source/user/library/examples.rst:979]: Done,

File: doc/source/user/library/examples.rst [hunk 6]
```diff
@@ -966,3 +966,318 @@ modules while in ``u.modules_stats`` the two dimension are switched.
...
+		The frozen bits, maybe generated from `py_generate_frozen_bits`
+	    info_bits : list (or ndarray) of information bits of shape (n_frame, k)
+		The information bits pending to be encoded
+	    Returns
+	    -------
+	    list of encoded bits, size (n_frame, n)
+		The polar encoded bits
+	    """
+	    assert info_bits.shape[1] > 0
+	    return polar_encode_multiple(k, n, frozen_bits, info_bits)
```
[the remark this answers is not in the document]
Comment (SubtleMuffin) [reply doc/source/user/library/examples.rst:1068]: I do not agree. Though this function is simple, but it provides a lot of performance gain, especially when user is trying to encode/decode multiple frames in Python. As you know, Python's for-loops are very expensive, not to mention we need to instantiate an encoder/decoder object every time we call it.

File: doc/source/user/library/examples.rst [hunk 7]
```diff
@@ -966,3 +966,318 @@ modules while in ``u.modules_stats`` the two dimension are switched.
...
+		The frozen bits, maybe generated from `py_generate_frozen_bits`
+	    received : list (or ndarray) of received LLRs, size (n_frame, n)
+		The received log-likelihood ratios (LLRs)
+	    Returns
+	    -------
+	    list of decoded bits, size (n_frame, k)
+		The polar decoded bits
+	    """
+	    assert received.shape[1] > 0
+	    return polar_decode_multiple(k, n, frozen_bits, received)
```
[the remark this answers is not in the document]
Comment (SubtleMuffin) [reply doc/source/user/library/examples.rst:1109]: No.

File: doc/source/user/library/examples.rst (hunk 1 above)
Comment (SubtleMuffin) [reply doc/source/user/library/examples.rst:1274]: Done.

File: doc/source/user/library/examples.rst (hunk 2 above)
Comment (SubtleMuffin) [reply doc/source/user/library/examples.rst:1199]: No.

File: doc/source/user/library/examples.rst (hunk 3 above)
Comment (SubtleMuffin) [reply doc/source/user/library/examples.rst:1247]: No.

File: doc/source/user/library/examples.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (kouchy) [reply doc/source/user/library/examples.rst:1067]: Ok, I see your point so perhaps you can tell in the text the purpose of the `multiple` function?

File: doc/source/user/library/examples.rst (hunk 2 above)
Comment (kouchy) [reply doc/source/user/library/examples.rst:1198]: I think there is too many blocks of code and not so many explanations. If the code is given with no comments, you can simply add a link to the file in the `my_project_with_aff3ct` repo. The other solution is to keep this code here but to split it in smaller blocks and to explain what you are doing in the text.

Comment (SubtleMuffin) [reply doc/source/user/library/examples.rst:1111]: Done. Please refer to the newest commit.

**Comment by SubtleMuffin** [comment]
Hey @kouchy , shall we wrap this up? Please let me know what else you would like to include!

**Comment by kouchy** [comment]
Hi @SubtleMuffin,

Sry, we had a new lock-down in my country, I did not have the time to focus on this yet, but I do not forget you :-).

Regards.

**Comment by SubtleMuffin** [comment]
Hey @kouchy : nothing is paramount than health. Please stay safe!

**Pull request closed without merging**